packages/cyberstorm/src/components/CopyButton/CopyButton.tsx

```diff
@@ -0,0 +1,72 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faClone } from "@fortawesome/free-regular-svg-icons";
+import { faCheck } from "@fortawesome/free-solid-svg-icons";
+import React, { PropsWithChildren, useEffect } from "react";
+import { useState } from "react";
+import styles from "./CopyButton.module.css";
+import { Tooltip } from "../Tooltip/Tooltip";
+
+type CopyButtonProps = {
+  text: string;
+};
+type CopiedValue = string | null;
+type CopyFn = (text: string) => Promise<void>;
+
+function useCopyToClipboard(): CopyFn {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [copiedText, setCopiedText] = useState<CopiedValue>(null);
+
+  const copy: CopyFn = async (text) => {
+    if (!navigator?.clipboard) {
+      console.warn("Clipboard not supported");
+    }
+
+    // Try to save to clipboard then save it in the state if worked
+    try {
+      await navigator.clipboard.writeText(text);
+      setCopiedText(text);
+    } catch (error) {
+      console.warn("Copy failed", error);
+      setCopiedText(null);
+    }
+  };
+
+  return copy;
+}
+
+export const CopyButton = React.forwardRef<HTMLButtonElement, CopyButtonProps>(
+  (props: PropsWithChildren<CopyButtonProps>) => {
+    const copy = useCopyToClipboard();
+    const [stateValue, setState] = useState(false);
+    useEffect(() => {
+      setTimeout(() => setState(false), 2000);
+    }, [stateValue]);
+    return (
+      <Tooltip content="Copy">
+        <button
```

Comment:
Would we want to use Cyberstorm's button component here, to allow styling the copy button like other buttons? I haven't checked the designs so I've no idea what we're aiming for. Just thinking aloud.

Reply:
Usually yes, that would be nice, but in this case the copybutton is its' own thing and differs from the cyberstorm Button's usage enough to be best done this way IMO.